eng/liveILLink.targets

```diff
@@ -0,0 +1,42 @@
+<Project>
+
+  <!-- NOTE: these targets should be a no-op, unless _RequiresLiveILLink, computed below, is true. -->
+
+  <PropertyGroup>
+    <!-- Keep these conditions in sync with _RequiresILLinkPack in
+         https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets -->
+    <_RequiresLiveILLink Condition="'$(_RequiresLiveILLink)' == '' And (
+        '$(PublishAot)' == 'true' Or
+        '$(IsAotCompatible)' == 'true' Or '$(EnableAotAnalyzer)' == 'true' Or
+        '$(PublishTrimmed)' == 'true' Or
+        '$(IsTrimmable)' == 'true' Or '$(EnableTrimAnalyzer)' == 'true' Or
+        '$(EnableSingleFileAnalyzer)' == 'true')">true</_RequiresLiveILLink>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(_RequiresLiveILLink)' == 'true'">
+    <_ILLinkTasksSourceDir>$(ToolsProjectRoot)illink\src\ILLink.Tasks\</_ILLinkTasksSourceDir>
+    <ILLinkAnalyzersPropsPath>$(ToolsProjectRoot)illink\src\ILLink.RoslynAnalyzer\build\Microsoft.NET.ILLink.Analyzers.props</ILLinkAnalyzersPropsPath>
+  </PropertyGroup>
+
+  <Import Project="$(_ILLinkTasksSourceDir)build\Microsoft.NET.ILLink.Tasks.props" Condition="'$(_RequiresLiveILLink)' == 'true'" />
+
+  <PropertyGroup Condition="'$(_RequiresLiveILLink)' == 'true'">
+    <!-- Don't use SDK's trimming functionality. -->
+    <_RequiresILLinkPack>false</_RequiresILLinkPack>
+    <ILLinkTasksAssembly>$(ToolsILLinkDir)$(NetCoreAppToolCurrent)\ILLink.Tasks.dll</ILLinkTasksAssembly>
```

Comment:
This needs to respect the MSBuildRuntimeType and pick the .NET Framework one, same as https://github.com/dotnet/runtime/blob/d3569b99ead8f909556f5b416b6f57d94f30e668/eng/illink.targets#L15-L17

Reply:
```suggestion
    <ILLinkTasksAssembly Condition="'$(MSBuildRuntimeType)' == 'Core'">$(ToolsILLinkDir)$(NetCoreAppToolCurrent)\ILLink.Tasks.dll</ILLinkTasksAssembly>
    <ILLinkTasksAssembly Condition="'$(MSBuildRuntimeType)' != 'Core'">$(ToolsILLinkDir)$(NetFrameworkToolCurrent)\ILLink.Tasks.dll</ILLinkTasksAssembly>
```